package.json

```diff
@@ -87,25 +87,28 @@
     "ts-loader": "^9.4.1",
     "tslib": "^2.4.1",
     "typescript": "^4.8.4",
-    "updeep": "^1.2.0",
+    "updeep": "^1.2.1",
     "val-loader": "^2.1.2",
     "webpack": "^5.36.2",
     "webpack-cli": "^4.7.2",
     "webpack-dev-middleware": "^4.3.0",
-    "webpack-hot-middleware": "^2.25.0",
+    "webpack-hot-middleware": "^2.25.3",
     "webpack-merge": "^5.7.3"
   },
   "scripts": {
-    "postinstall": "lerna bootstrap && link-parent-bin && link-parent-bin -c package-templates && yarn run tryBuild",
-    "check": "npm run lint && npm run test",
     "reinstall": "rm -rf node_modules yarn.lock && yarn",
     "ci": "yarn install --frozen-lockfile",
     "build": "lerna exec --parallel --no-bail yarn run build",
     "tryBuild:preview": "lerna exec --parallel --no-bail yarn run build:preview || true",
     "tryBuild": "lerna exec --parallel --no-bail yarn run build || true",
-    "lint": "npm run eslint && npm run stylelint",
-    "eslint": "eslint -f stylish \"./**/*.{js,jsx,ts,tsx}\" ",
-    "stylelint": "stylelint \"./**/*.scss\" ",
-    "test": "scripts/run-tests"
+    "--- CHECKS AND TEST SCRIPTS ---": "-----------------------------",
+    "check": "npm test && npm run lint && echo \"\\033[0;32mAll checks pass\\033[0m\"",
+    "lint": "npm run lint:javascript && npm run lint:styles && npm run lint:typescript",
+    "lint:javascript": "eslint -f stylish \"./**/*.{js,jsx,ts,tsx}\"",
+    "lint:typescript": "tsc --noEmit",
```

Comment:
really wonder if this works like that since every package might have their own tsconfig file (extending the root one). pretty similar to tests

Reply:
Adding an error (like removing a type from https://github.com/goodhood-eu/goodhood/blob/9f973714bb82f80f4f8b9705acbed6d8b2ee021b/packages/modals/src/modal/hooks.tsx#L24) will trigger an error when check is called from the root package.json as well as when calling "build" from the modals-packages scripts.

Or what was the desired behaviour / rule for checking ts?

Reply:
Pretty sure it works currently because of the changes you did to root `tsconfig.json` but we have to acknowledge the package `tsconfig.json` file (eg. packages/modals/tsconfig.json). This is the one that actually counts. Some package could also decide on going for a full custom config (pretty much the same thing with building)

Suggestion for solution:
- Add `lint:typescript` script to every package (executing `scripts/check-typescript`, running `tsc --noEmit`)
- Add `lint:typescript` script to root package, running `lint:typescript` for every package (same as `build`)

Reply:
added those things...